elixir/apps/domain/lib/domain/tokens/jobs/refresh_browser_session_tokens.ex

```diff
@@ -0,0 +1,40 @@
+defmodule Domain.Tokens.Jobs.RefreshBrowserSessionTokens do
+  use Domain.Jobs.Job,
+    otp_app: :domain,
+    every: :timer.minutes(5),
+    executor: Domain.Jobs.Executors.GloballyUnique
+
+  alias Domain.Repo
+  alias Domain.Auth
+  alias Domain.Tokens
+  require Logger
+
+  @impl true
+  def execute(_config) do
+    Tokens.all_active_browser_session_tokens!()
+    |> Repo.preload(identity: :provider)
+    |> Enum.each(fn token ->
+      with {:ok, _identity, expires_at} when not is_nil(expires_at) <-
+             Auth.Adapters.refresh_access_token(token.identity.provider, token.identity),
+           {:ok, _token} <- Tokens.update_token(token, %{expires_at: expires_at}) do
+        :ok
+      else
+        {:ok, _identity, nil} ->
+          :ok
+
+        {:error, :not_supported} ->
+          :ok
+
+        {:error, reason} ->
+          Logger.warning("Failed to refresh browser session token",
```

Comment:
Just so I understand when this could happen:

- User is removed / suspended from IdP since signing in
- Connector is removed since signing in
- User is deleted / demoted by another admin

Is that right? Any other cases I'm missing?

Reply:
@jamilbk also just due to network timeouts. Worst thing that will happen is that session will not be extended.